File: apps/backend/app/schemas/task.py
from pydantic import BaseModel, ConfigDict
from typing import Optional, List, Dict, Any
from datetime import date, time, datetime
from app.models.task import TaskPriority, TaskStatus

class SubtaskCreate(BaseModel):
    title: str
    sort_order: int = 0

class SubtaskResponse(BaseModel):
    id: int
    task_id: int
    title: str
    completed: bool
    sort_order: int
    
    model_config = ConfigDict(from_attributes=True)

class TaskCreate(BaseModel):
    title: str
    description: Optional[str] = None
    priority: TaskPriority = TaskPriority.media
    category: Optional[str] = None
    due_date: Optional[date] = None
    due_time: Optional[time] = None
    recurrence: Optional[str] = None
    tags: Optional[Dict[str, Any]] = None
    parent_task_id: Optional[int] = None
    subtasks: Optional[List[SubtaskCreate]] = None

class TaskUpdate(BaseModel):
    title: Optional[str] = None
    description: Optional[str] = None
    priority: Optional[TaskPriority] = None
    status: Optional[TaskStatus] = None
    category: Optional[str] = None
    due_date: Optional[date] = None
    due_time: Optional[time] = None
    recurrence: Optional[str] = None
    tags: Optional[Dict[str, Any]] = None

class TaskResponse(BaseModel):
    id: int
    title: str
    description: Optional[str] = None
    priority: TaskPriority
    status: TaskStatus
    category: Optional[str] = None
    due_date: Optional[date] = None
    due_time: Optional[time] = None
    recurrence: Optional[str] = None
    tags: Optional[Dict[str, Any]] = None
    parent_task_id: Optional[int] = None
    completed_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
    subtasks: List[SubtaskResponse] = []
    
    model_config = ConfigDict(from_attributes=True)

class TaskSummary(BaseModel):
    total: int
    completed: int
    pending: int
    overdue: int
    by_priority: Dict[str, int]

File: apps/desktop/src/types/task.ts
// Task types matching backend schema

export type Priority = "baixa" | "media" | "alta" | "urgente";
export type TaskStatus = "pendente" | "em_andamento" | "concluida" | "arquivada";
export type Recurrence = "none" | "daily" | "weekly" | "monthly";

export interface Task {
  id: number;
  title: string;
  description: string | null;
  priority: Priority;
  status: TaskStatus;
  category: string | null;
  due_date: string | null;
  due_time: string | null;
  recurrence: Recurrence;
  tags: string[];
  parent_task_id: number | null;
  completed_at: string | null;
  created_at: string;
  updated_at: string;
  subtasks: Subtask[];
}

export interface Subtask {
  id: number;
  task_id: number;
  title: string;
  completed: boolean;
  sort_order: number;
  created_at: string;
}

export interface TaskCreate {
  title: string;
  description?: string;
  priority?: Priority;
  status?: TaskStatus;
  category?: string;
  due_date?: string;
  due_time?: string;
  recurrence?: Recurrence;
  tags?: string[];
  parent_task_id?: number;
  subtasks?: { title: string }[];
}

export interface TaskUpdate {
  title?: string;
  description?: string;
  priority?: Priority;
  status?: TaskStatus;
  category?: string;
  due_date?: string;
  due_time?: string;
  recurrence?: Recurrence;
  tags?: string[];
}

export interface TaskSummary {
  total: number;
  pending: number;
  in_progress: number;
  completed: number;
  overdue: number;
}

export type TaskFilter = "all" | "today" | "overdue" | "high_priority" | "completed";
